h/static/scripts/directive/test/util.js

```diff
@@ -151,6 +151,12 @@ function createDirective(document, name, attrs, initialScope, initialHtml, opts)
     element.scope = childScope;
     childScope.$digest();
     element.ctrl = element.controller(name);
+
+    if (!element.ctrl) {
```

Comment:
I added this because it bit me a few times. This is also why no-op controllers were added to 3 components.